### PR TITLE
Feature/106 parse data types

### PR DESCRIPTION
**Pull request by bergnerjonas**
This is a quick fix for the date parser in zensus, which uses YYYY-MM-DD regardless of language settings.

## Summary by CodeRabbit

- **New Features**
	- Improved date format handling for data retrieval based on language and database name, enhancing user experience when working with date strings.
  
- **Bug Fixes**
	- Resolved inconsistencies in date format for the "zensus" database when the language is set to German.

### CODE REVIEW

**Comment by pmayd** [comment]
@bergnerjonas what is the current status of this?

**Comment by bergnerjonas** [comment]
> @bergnerjonas what is the current status of this?

Ready to merge!

File: src/pystatis/table.py [hunk 1]
```diff
@@ -154,7 +154,9 @@ def get_data(
             decimal="," if language == "de" else ".",
             dtype={raw_data_header.split(";")[pos + 2]: str for pos in pos_of_ags_col},
             parse_dates=[config.LANG_TO_COL_MAPPING[db_version][language]["time"]],
-            date_format="%d.%m.%Y" if language == "de" else "%Y-%m-%d",
+            date_format="%d.%m.%Y"
+            if (language == "de" and db_name != "zensus")
```
Comment (pmayd) [inline src/pystatis/table.py:158]: are you sure this is still correct? Please check with current Genesis version as they switch to the new database version that is like Zensus format so I would actually expect they have the same format now?

**Review by pmayd — COMMENTED** [review]
Could you check pls?

File: src/pystatis/table.py (hunk 1 above)
Comment (bergnerjonas) [reply src/pystatis/table.py]: Fixed this so genesis and zensus now both use the new format for either language. Added tests for these.

**Review by pmayd — APPROVED** [review]
LGTM, thanks

**Pull request merged**